cmd/app/unlink.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022-2025 Salesforce, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/slackapi/slack-cli/internal/cmdutil"
+	"github.com/slackapi/slack-cli/internal/iostreams"
+	"github.com/slackapi/slack-cli/internal/prompts"
+	"github.com/slackapi/slack-cli/internal/shared"
+	"github.com/slackapi/slack-cli/internal/shared/types"
+	"github.com/slackapi/slack-cli/internal/slacktrace"
+	"github.com/slackapi/slack-cli/internal/style"
+	"github.com/spf13/cobra"
+)
+
+// Handle to function used for testing
+var unlinkAppSelectPromptFunc = prompts.AppSelectPrompt
+
+// NewUnlinkCommand returns a new Cobra command for unlinking apps
+func NewUnlinkCommand(clients *shared.ClientFactory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "unlink",
+		Short: "Remove linked app from the project",
+		Long:  "Unlink a previously linked app from the project",
+		Example: style.ExampleCommandsf([]style.ExampleCommand{
+			{
+				Meaning: "Remove an existing app from the project",
+				Command: "app unlink",
+			},
+			{
+				Meaning: "Remove a specific app without using prompts",
+				Command: "app unlink --team T0123456789 --app A0123456789 --environment deployed",
+			},
+		}),
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			clients.Config.SetFlags(cmd)
```

Comment:
```suggestion
		PreRunE: func(cmd *cobra.Command, args []string) error {
```

🪓 quibble: Unsure if this is required since additional flags aren't being set here. However, IIRC we have a longstanding hope of making the "app" flag command specific since it's not useful for all commands such as `login` and similar!